mlflow/data/pandas_dataset.py

```diff
@@ -0,0 +1,149 @@
+import hashlib
+import json
+from typing import Optional, Any, Dict, Union
+
+import numpy as np
+import pandas as pd
+from functools import cached_property
+
+from mlflow.data.dataset import Dataset
+from mlflow.data.filesystem_dataset_source import FileSystemDatasetSource
+from mlflow.data.spark_dataset_source import SparkDatasetSource
+from mlflow.data.delta_dataset_source import DeltaDatasetSource
+from mlflow.data.dataset_source import DatasetSource
+
+from mlflow.data.pyfunc_dataset_mixin import PyFuncConvertibleDatasetMixin, PyFuncInputsOutputs
+from mlflow.types import Schema
+from mlflow.types.utils import _infer_schema
+
+
+class PandasDataset(Dataset, PyFuncConvertibleDatasetMixin):
+    """
+    Represents a Pandas dataset for use with MLflow Tracking.
+    """
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        source: Union[FileSystemDatasetSource, SparkDatasetSource, DeltaDatasetSource],
+        targets: str = None,
```

Comment:
Can we add a `targets` method to this class that returns the name of the targets column? For reference, here's the definition from the HuggingFace dataset:

```
    @property
    def targets(self) -> Optional[str]:
        """
        The name of the Hugging Face dataset column containing targets.

        :return: The Hugging Face `datasets.Dataset` instance.
        """
        return self._targets
```